lib/volcanic/authenticator/v1/subject.rb

```diff
@@ -23,15 +23,27 @@ def privileges_for(subject, service)
             exception,
             "#{path}/privileges?filter[subject]=#{subject}&filter[service]=#{service}"
           )
-          return [] if response.blank?
+          privilege_parser(response)
+        end
+
+        # similar to +privileges_for+, accept it
+        # search privileges by token scope
+        def privileges_for_scope(token_base64, service)
```

Comment:
`privileges_for_scope` is a confusing name.... we're not getting the privileges related to the scope, but to the specified token. A better name would be `privileges_for_token`.

Given that this has nothing to do with the subject, the method is probably in the wrong place... It is probably code that should be in the `Token` class

Reply:
both method `privileges_for_scope` and `privileges_for` is calling the same resource path which is `'api/v1/subject'`. That is why thought of applying it at `subject` class.